sysdeps/linux/generic/entry.cpp

```diff
@@ -26,7 +27,8 @@ LibraryGuard::LibraryGuard() {
 			__mlibc_stack_data.envp);
 }
 
-extern "C" void __mlibc_entry(int (*main_fn)(int argc, char *argv[], char *env[])) {
+extern "C" void __mlibc_entry(int (*main_fn)(int argc, char *argv[], char *env[]), uintptr_t *entry_stack) {
```

Comment:
Minor: IMHO, inverting the order of the arguments (`entry_stack` first) makes more sense.